### ISSUE TITLE
Python above version 3.9 is not supported

### ISSUE THREAD

**Issue by nmnsnv** [nmnsnv/regexp_sar#5]
We fail to run SAR on Python versions above 3.9 because of this and receive the following error

```
>>> sar.add_regexp(".+", lambda f, t: print(f"Match!"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<SOME-PATH>/python3.11/site-packages/regexp_sar/__init__.py", line 13, in add_regexp
    _regexp_sar_c.add_regexp(self._internal_sar_object, regexp, callback)
SystemError: PY_SSIZE_T_CLEAN macro must be defined for '#' formats
```

This is caused because of Pythons C-API change for versions above 3.9.
Per [Pythons C-API docs](https://docs.python.org/3/c-api/arg.html):

```
For all # variants of formats (s#, y#, etc.), the macro PY_SSIZE_T_CLEAN must be defined before including Python.h. On Python 3.9 and older, the type of the length argument is [Py_ssize_t](https://docs.python.org/3/c-api/intro.html#c.Py_ssize_t) if the PY_SSIZE_T_CLEAN macro is defined, or int otherwise.
```

**Comment by nmnsnv** [comment]
[Created Fix for this PR](#6)

**Comment by nmnsnv** [comment]
The issue was fixed by updating Pythons API and defining PY_SSIZE_T_CLEAN.
New version released.
